find/handler.go

```diff
@@ -51,23 +57,65 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		q.Set("query", query)
 		r.URL.RawQuery = q.Encode()
 	} else {
+		switch r.FormValue("format") {
+		case "json":
+		case "pickle":
+		case "protobuf":
+		default:
+			logger.Error("unsupported formatter")
+			http.Error(w, "Failed to parse request: unsupported formatter", http.StatusBadRequest)
+			return
+		}
 		query = r.FormValue("query")
 	}
 	if len(query) == 0 {
 		http.Error(w, "Query not set", http.StatusBadRequest)
 		return
 	}
+
+	var key string
+	useCache := h.config.Common.FindCache != nil && h.config.Common.FindCacheConfig.FindTimeoutSec > 0 && !parser.TruthyBool(r.FormValue("noCache"))
+	if useCache {
+		// logger = logger.With(zap.String("use_cache", "true"))
+		ts := utils.TimestampTruncate(time.Now().Unix(), time.Duration(h.config.Common.FindCacheConfig.FindTimeoutSec)*time.Second)
+		key = "1970-02-12;query=" + query + ";ts=" + strconv.FormatInt(ts, 10)
+		body, err := h.config.Common.FindCache.Get(key)
+		if err == nil {
+			// ApiMetrics.RequestCacheHits.Add(1)
```

Comment:
I'm not sure why that is commented out.

Reply:
Can be removed - some metrics (gc/cache hits-miss/query stats) planed in future

Reply:
May be sended in statsd or in-place aggregate